src/gsy_e/models/strategy/energy_parameters/heatpump/tank.py

```diff
@@ -1,219 +1,106 @@
+        for tank in self._tanks_states:
             tank.decrease_tank_temp_from_heat_energy(heat_energy_per_tank_kWh, time_slot)
 
     def update_tanks_temperature(self, time_slot: DateTime):
         """
         Update the current temperature of all tanks, based on temp increase/decrease of the market
         slot.
         """
-        for tank in self._tanks_energy_parameters:
+        for tank in self._tanks_states:
             # pylint: disable=protected-access
-            tank._state.update_storage_temp(time_slot)
+            tank.update_storage_temp(time_slot)
 
-    def get_max_energy_consumption(self, cop: float, time_slot: DateTime):
-        """Get max energy consumption from all water tanks."""
+    def get_max_heat_energy_consumption_kJ(self, time_slot: DateTime):
+        """Get max heat energy consumption from all water tanks."""
         max_energy_consumption_kWh = sum(
-            tank.get_max_energy_consumption(cop, time_slot)
-            for tank in self._tanks_energy_parameters
+            tank.get_max_heat_energy_consumption_kWh(time_slot) for tank in self._tanks_states
         )
         assert max_energy_consumption_kWh > -FLOATING_POINT_TOLERANCE
-        return max_energy_consumption_kWh
+        return convert_kWh_to_kJ(max_energy_consumption_kWh)
 
-    def get_min_energy_consumption(self, cop: float, time_slot: DateTime):
-        """Get min energy consumption from all water tanks."""
-        if cop == 0:
-            return 0
+    def get_min_heat_energy_consumption_kJ(self, time_slot: DateTime):
+        """Get min heat energy consumption from all water tanks."""
         min_energy_consumption_kWh = sum(
-            tank.get_min_energy_consumption(cop, time_slot)
-            for tank in self._tanks_energy_parameters
+            tank.get_min_heat_energy_consumption_kWh(time_slot) for tank in self._tanks_states
         )
         assert min_energy_consumption_kWh > -FLOATING_POINT_TOLERANCE
-        return min_energy_consumption_kWh
+        return convert_kWh_to_kJ(min_energy_consumption_kWh)
 
     def get_average_tank_temperature(self, time_slot: DateTime):
         """Get average tank temperature of all water tanks."""
-        return mean(
-            tank.current_tank_temperature(time_slot) for tank in self._tanks_energy_parameters
-        )
+        return mean(tank.current_tank_temperature(time_slot) for tank in self._tanks_states)
 
     def get_unmatched_demand_kWh(self, time_slot: DateTime):
         """Get unmatched demand of all water tanks."""
-        return sum(
-            tank.get_unmatched_demand_kWh(time_slot) for tank in self._tanks_energy_parameters
-        )
+        return sum(tank.get_unmatched_demand_kWh(time_slot) for tank in self._tanks_states)
 
     def serialize(self) -> Union[Dict, List]:
         """Serializable dict with the parameters of all water tanks."""
         # TODO: Convert the AVRO schemas to be able to serialize multiple tanks
```

Comment:
You did that in this PR: https://github.com/gridsingularity/gsy-framework/pull/595
Right?
If so, you can remove this comment. :-)

Reply:
This is still pending and no good way to resolve this comes to mind :( This needs changes in the simulation results service in order to be able to save the tanks results as a list. Let me give it a quick try just in case it is not breaking too many things.